### DISCUSSION TITLE
Switching to weld-testing

### DISCUSSION THREAD

**Discussion by Nerdy-5** [cdi-unit/cdi-unit#222] [General]
Hi folks!

After using cdi-unit for many years, it looks like the project has gone stale and won't be fit für Java 17. I just stumbled upon weld-testing https://github.com/weld/weld-testing/ and it looks promising, with nearly identical features and more.

Before jumping ship, I'd like to inquire about the experience others made.
Were you able to upgrade zu Java 17?
Did you get rid of the transitive vulnerabilities cdi-unit pulls into your project?
Did you switch to weld-testing or some other framework?
Did you encounter many problems?

Best regards, Thomas

**Comment by ikysil** [comment]
hi @Nerdy-5 

Yeah, the work for Jakarta EE support is still in progress - see #195 for details.

What version of JakartaEE/Weld are you targeting?

**Reply by Nerdy-5** [reply]
First I'll try to make it work with Weld 3.x, because, that's what we still use and then will be the big upgrade to the newest versions.
Already hit a snag with migrating to weld-testing, but I'm sure I'll figure it out eventually.

**Comment by ikysil** [comment]
Version 5.0.0-EA1 has been published.
[`pkg:maven/io.github.cdi-unit/cdi-unit@5.0.0-EA1`](https://central.sonatype.com/artifact/io.github.cdi-unit/cdi-unit/5.0.0-EA1)

Please check and report any issues!
